gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/JobSpec.java

```diff
@@ -289,6 +298,23 @@ public Builder withTemplate(URI templateURI) {
       this.templateURI = Optional.of(templateURI);
       return this;
     }
+
+    public Map getDefaultMetadata() {
+      return ImmutableMap.of(VERB_KEY, SpecExecutor.Verb.ADD.name());
```

Comment:
Why have ADD verb in the default metadata? Maybe the build should raise an error if the metadata is not present or some required keys like the verb are missing.

Reply:
As some specs/test do not need it, I've added UNKNOWN as a default verb to make it backward compatible.